### PR TITLE
Update documentation and fix typos

### PR DESCRIPTION
**Pull request by nmattia**
These are a fixes for a few problems I spotted while setting up the
environment and going through the II spec. If anything's unclear or
incorrect it's probably because I didn't understand it :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,9 +45,13 @@ II_ENV=development dfx deploy --no-wallet --argument '(null)'
 Then the canister can be used as
 
 ```bash
-dfx canister call internet_identity register '(123, "test", vec {1; 2; 3}, null)'
+$ dfx canister call internet_identity init_salt
```
Comment (nmattia) [inline README.md:48]: The old call was probably using an old API. `init_salt` is the closest thing I could find to a health check, where the API is unlikely to break.

File: docs/internet-identity-spec.adoc [hunk 2]
```diff
@@ -132,9 +132,9 @@ where `msg` is a value with of type
 where
 
 * the `sessionPublicKey` contains the public key of the session key pair.
-* the `maxTimeToLive`, if present, indicates the desired time span until the requested delegation should expire. The Identity Provider frontend is free to set an earlier expiry time, but should not create a larger.
+* the `maxTimeToLive`, if present, indicates the desired time span (in nanoseconds) until the requested delegation should expire. The Identity Provider frontend is free to set an earlier expiry time, but should not create a larger.
```
Comment (nmattia) [inline docs/internet-identity-spec.adoc:135]: Wasn't sure about this one, but other time spans in the doc are expressed as nanos.

File: docs/internet-identity-spec.adoc [hunk 3]
```diff
@@ -342,8 +342,8 @@ Therefore the Internet Identity Canister intentionally does not implement `canis
 
 The primary data structure used by the backend is a map from Identity Anchor to the list of user devices.
 Device lists are stored directly in canister stable memory.
-The total amount of storage for is limited to 510 bytes per user.
-With the stable memory size of 4GiB we can store around 8 * 10^6 user records in a single canister.
+The total amount of storage for is limited to 2k bytes per user.
```
Comment (nmattia) [inline docs/internet-identity-spec.adoc:345]: Mentioned by @kritzcreek but I didn't check the math.

File: README.md [hunk 4]
```diff
@@ -39,26 +39,56 @@ To run the internet_identity canisters, proceed as follows after cloning the rep
...
+To serve the frontend locally via webpack (recommended during development), run
+the following:
 
 ```bash
-echo "http://localhost:8000?canisterId=$(dfx canister id internet_identity)"
+CANISTER_ID=$(dfx canister id internet_identity) npm start
+open "http://localhost:8080"
 ```
 
+Finally, to test workflows like authentication from a client application, you start the sample app:
```
Comment (kritzcreek) [inline README.md:75]: I think this section should go under `Contributing to the frontend`, where you have the TODO right now

File: docs/internet-identity-spec.adoc (hunk 3 above)
Comment (kritzcreek) [reply docs/internet-identity-spec.adoc:345]: It's 2kb as specified here: https://github.com/dfinity/internet-identity/blob/5f5bd98e44f89ebd321a5c3b3d5ee36d7cd4a799/src/internet_identity/src/storage.rs#L12

**Review by kritzcreek — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (nmattia) [reply README.md:90]: Agreed

**Comment by nmattia** [comment]
Thanks! PTAL.

File: docs/internet-identity-spec.adoc (hunk 3 above)
Comment (nmattia) [reply docs/internet-identity-spec.adoc]: Updated to 2KiB

File: README.md [hunk 5]
```diff
@@ -64,13 +74,30 @@ dfx start [--clean] [--background]
 II_ENV=development dfx deploy --no-wallet --argument '(null)'
 ```
 
-Then, run `CANISTER_ID=$(dfx canister id internet_identity) npm start` to start webpack-dev-server.
+To serve the frontend locally via webpack (recommended during development), run
+the following:
+
+```bash
+CANISTER_ID=$(dfx canister id internet_identity) npm start
+open "http://localhost:8080"
```
Comment (kritzcreek) [inline README.md:82]: Same here, `open` is OSX specific.

File: README.md [hunk 6]
```diff
@@ -39,19 +39,29 @@ To run the internet_identity canisters, proceed as follows after cloning the rep
...
 ```
 
-To open the front-end, you can run the following and open the URL.
+See `dfx canister call --help` and [the documentation](https://sdk.dfinity.org/docs/developers-guide/cli-reference/dfx-canister.html#_examples) for more information.
+
+The `dfx` executable can proxy queries to the canister. To view it, run the following:
 
 ```bash
-echo "http://localhost:8000?canisterId=$(dfx canister id internet_identity)"
+open "http://localhost:8000?canisterId=$(dfx canister id internet_identity)"
```
Comment (kritzcreek) [inline README.md:64]: `open` is a MacOS specific command, that I'd rather not include here. On Linux it's `xdg-open`. I prefer the previous instruction to just open the URL however people do that on their machine.

File: README.md [hunk 7]
```diff
@@ -64,13 +74,30 @@ dfx start [--clean] [--background]
...
 ```
 CANISTER_ID=rrkah-fqaaa-aaaaa-aaaaq-cai
 ```
 
+Finally, to test workflows like authentication from a client application, you start the sample app:
+
+```bash
+cd demos/sample-javascript
+npm run develop
+open "http://localhost:8081"
```
Comment (kritzcreek) [inline README.md:95]: One more `open`

File: README.md [hunk 8]
```diff
@@ -64,13 +74,30 @@ dfx start [--clean] [--background]
 II_ENV=development dfx deploy --no-wallet --argument '(null)'
 ```
 
-Then, run `CANISTER_ID=$(dfx canister id internet_identity) npm start` to start webpack-dev-server.
+To serve the frontend locally via webpack (recommended during development), run
+the following:
+
+```bash
+CANISTER_ID=$(dfx canister id internet_identity) npm start
```
Comment (kritzcreek) [inline README.md:81]: I don't need to do the `CANISTER_ID` bit on my end. If you've run `dfx deploy` beforehand (to install the canister) the canisterId should be available and picked up by the webpack config from `.dfx/local/canister_ids.json` automatically.

**Review by kritzcreek — CHANGES_REQUESTED** [review]
Sorry, I missed the calls to `open` in my previous review.

File: README.md (hunk 8 above)
Comment (nmattia) [reply README.md:81]: You're correct. Amending.

**Comment by nmattia** [comment]
Thanks @kritzcreek ! PTAL.

**Review by kritzcreek — APPROVED** [review]
Thanks!

**Pull request merged**